crowbar_framework/app/models/service_object.rb

```diff
@@ -1279,6 +1395,8 @@ def apply_role(role, inst, in_queue)
       end
 
       # Make sure the config role is on the nodes in this barclamp, otherwise remove it
+      # FIXME: All nodes is basically all new nodes, which should have an add/remove list,
+      # why do we need to add specifically the passed role?
```

Comment:
In general there are *three* sets of roles, because there are also node roles.  This is explained by slide 65 of https://w3.suse.de/~aspiers/cloud/HA-training/HA-training-part-2a.pdf